pkg/provider/aad/aad.go

```diff
@@ -670,8 +671,9 @@ func (ac *Client) getJsonFromConfig(resBodyStr string) string {
 	 * <script><![CDATA[  $Config=......; ]]>
 	 */
 	startIndex := strings.Index(resBodyStr, "$Config=") + 8
-	endIndex := startIndex + strings.Index(resBodyStr[startIndex:], ";")
-	return resBodyStr[startIndex:endIndex]
+	endIndex := startIndex + strings.Index(resBodyStr[startIndex:], "//]]>") - 1
+	untrimmedJsonStr := resBodyStr[startIndex:endIndex]
+	return untrimmedJsonStr[:strings.LastIndex(untrimmedJsonStr, "}")+1]
```

Comment:
Why don’t we take advantage of the fact that a streaming JSON parser will return the first parsed JSON entity and ignore any additional data following it:
```
var msg json.RawMessage
_ = json.NewDecoder(strings.NewReader(resBodyStr[startIndex:])).Decode(&msg)
return string(msg)
```

Otherwise, the code as proposed could break if there are additional lines of javascript, e.g. `$Config={...}; if (false) {  alert("oh noes!") };` Best of all, _no more ad hoc self parsing_.